### PR TITLE
Use mariadb_config binary to determine compiler flags

### LINKED ISSUE #15: Building on macOS fails
Hi,

I am trying to build things on macOS and the configure script fails. This is due to neither `mysqlclient` nor `mariadb` usually being in the library path (at least not when installing the connector via `brew`). The way to build things is to use `mysql_config --cflags` or `mariadb_config --cflags`, which will return the correct locations for building on both macOS as well as GNU/Linux systems, so I think it is better to use that than to attempt to compile a `detect.c`.

### PR DESCRIPTION
**Pull request by Leonidas-from-XIV**
This will use `mariadb_config` to determine what to compile and link against. Closes #15.

I've removed support for switching between connector and uhm, server, since that could be handled by putting different `mariadb_config` in the path.

I've let the `mariadb`/`mysql` include prefix in, since I think it would be nice to be able to compile against the MySQL connector, but for this PR I only added `mariadb_config` support and ignore `mysql_config`. I am not sure this library is compatible with building against MySQL connector anyway.

### CODE REVIEW

File: myocamlbuild.ml [hunk 1]
```diff
@@ -1206,14 +1216,14 @@ let conf = {MyOCamlbuildFindlib.no_automatic_syntax = false}
...
 
-# 1210 "myocamlbuild.ml"
+# 1220 "myocamlbuild.ml"
 (* OASIS_STOP *)
 
 let dispatch = function
   | After_rules ->
     let stubgen          = "stubgen/ffi_stubgen.byte" in
     let stubgen_types    = "stubgen/ffi_types_stubgen.byte" in
-    let stubgen_ml_types = "stubgen/ffi_ml_types_subgen" in
```
Comment (Leonidas-from-XIV) [inline myocamlbuild.ml]: This is an unrelated change and I am quite surprised how this used to work.

File: myocamlbuild.ml [hunk 2]
```diff
@@ -1206,14 +1216,14 @@ let conf = {MyOCamlbuildFindlib.no_automatic_syntax = false}
...
+    let stubgen_ml_types = "stubgen/ffi_ml_types_stubgen" in
```
Comment (Leonidas-from-XIV) [inline myocamlbuild.ml]: This is an unrelated change and I am quite surprised how this used to work.

**Comment by andrenth** [comment]
Hi

This is not correct. OCaml-MariaDB supports MariaDB's two connectors:

1) MariaDB C/Connector (libmariadb-dev): link with `-lmariadb` and include `mariadb/mysql.h`
2) MaraDB's libmysqlclient (libmariadbclient-dev): link with `-lmysqlclient` and include `mysql/mysql.h`

For the second case, you must use `mysql_config` to get the flags.

I'd like to support both connectors at least until an Ubuntu LTS version has a C/Connector package that supports the nonblocking APIs.

**Comment by Leonidas-from-XIV** [comment]
Oh, I see. I'll add support for `mysql_config` as well. Probably tomorrow.

**Comment by Leonidas-from-XIV** [comment]
My MariaDB connector from `brew` does not come with a `mysql_config` and using the one provided by the MySQL connector does not work since there is no `mysql/mysql.h` but I suppose this is just because of weird packaging.

I was actually hoping to be able to use this to connect to an (actual) MySQL server but that seems to fail with all kinds of errors.

**Comment by andrenth** [comment]
Doesn't the MariaDB brew package come with a `mysql_config`?

**Comment by paurkedal** [comment]
Since I switched to use dune and dune-configurator (#52), this PR does not apply any more, and I'm wondering if we can close it, given that #15 was resolved?

The dune switch replicated the existing functionality, rather than using `mysql_config` and `mariadb_config`. The latter can be implemented, but given there are two commands and if the former does not always give use the flags we need, I suggest we only resurrect this PR if it still solves a real problem.

**Comment by Leonidas-from-XIV** [comment]
Yeah, I don't use MySQL nor MariaDB anymore, so I'm have no need for it either way.
